core/state/state_object.go

```diff
@@ -468,22 +597,59 @@ func (s *stateObject) setNonce(nonce uint64) {
 	s.data.Nonce = nonce
 }
 
-// Quorum - Privacy Enhancements
-func (s *stateObject) SetStatePrivacyMetadata(metadata *PrivacyMetadata) {
-	prevPM, _ := s.PrivacyMetadata()
-	s.db.journal.append(privacyMetadataChange{
+// Quorum
+// SetAccountExtraData modifies the AccountExtraData reference and journals it
+func (s *stateObject) SetAccountExtraData(extraData *AccountExtraData) {
+	current, _ := s.AccountExtraData()
+	s.db.journal.append(accountExtraDataChange{
+		account: &s.address,
+		prev:    current,
+	})
+	s.setAccountExtraData(extraData)
+}
+
+// A new AccountExtraData will be created if not exists.
+// This must be called after successfully acquiring accountExtraDataMutex lock
+func (s *stateObject) journalAccountExtraData() *AccountExtraData {
+	current, _ := s.AccountExtraData()
+	s.db.journal.append(accountExtraDataChange{
 		account: &s.address,
-		prev:    prevPM,
+		prev:    current.copy(),
```

Comment:
Wouldn't `copy()` fail if current was nil? Maybe do the `nil` test below before the `copy()`.

Reply:
not needed as `copy()` will return `nil` if the target is `nil`

https://github.com/trung/quorum/blob/64f8fa18967d3b50dbf8e52a9a1d6674d7ed8bf3/core/state/state_object.go#L232-L235

I'm adding a test case to reflect the same